keyserver/internal/internal.go

```diff
@@ -344,6 +378,23 @@ func (a *KeyInternalAPI) queryRemoteKeys(
 				res.DeviceKeys[userID][deviceID] = keyJSON
 			}
 		}
+
+		for userID, body := range result.MasterKeys {
+			switch b := body.CrossSigningBody.(type) {
+			case *gomatrixserverlib.CrossSigningKey:
```

Comment:
Why do we need to type cast here? Why can't `body` just specify the correct types?

Reply:
Because of [this](https://github.com/matrix-org/gomatrixserverlib/blob/master/crosssigning.go#L57). If there's a `device_id` field in the JSON then we actually unmarshal into a `gomatrixserverlib.DeviceKeys` (signing a device), otherwise we unmarshal into a `gomatrixserverlib.CrossSigningKey` (signing a key).